pkg/controller/workload/kubernetes/resource/resource.go

```diff
@@ -311,8 +315,10 @@ func (c *unstructuredClient) delete(ctx context.Context, template *unstructured.
 		return nil
 	}
 
-	// The object exists and we own it. Delete it.
-	return errors.Wrap(c.kube.Delete(ctx, remote), errDeleteResource)
+	if err := c.kube.Delete(ctx, remote); err != nil {
+		return errors.Wrap(err, errDeleteSecret)
+	}
+	return errors.New(errWaitingForDeletion)
```

Comment:
I'm not a huge fan of using an error as control flow for a seemingly non-error condition here. Would it be a lot riskier or more work to change the signature of these types to return some information? Perhaps:

```go
type Deletion struct {
    Done bool
}

# Maybe just (bool, error) if we don't think we're likely to need to return more information in future?
func (c *unstructuredClient delete(ctx context.Context, template *unstructured.Unstructured) (Deletion, error) {
  # ...
}
```

Reply:
I just went for the cheapest option but will look at improving it after testing that it solves the problem.

Reply:
Tested and it does block KAR deletion until remote resources are gone.

> but will look at improving it after testing that it solves the problem.

@negz In the whole tree of calls, there is the assumption that deletion is made once and as long as it doesn't fail, it succeeds and it succeeds immediately. Since this will go into a patch release, I'd prefer shipping it like this and do another iteration for 0.12. Maybe a big refactor of this KA and KAR controllers.